addons/web/static/tests/views/fields/badge_selection_with_filter_field.test.js

```diff
@@ -0,0 +1,201 @@
+import { expect, test } from "@odoo/hoot";
+import { click } from "@odoo/hoot-dom";
+import { animationFrame } from "@odoo/hoot-mock";
+import { defineModels, fields, models, mountView } from "@web/../tests/web_test_helpers";
+
+class Partner extends models.Model {
+    is_raining_outside = fields.Boolean();
+    mood = fields.Selection({
+        selection: [
+            ["happy", "Happy"],
+            ["sad", "Sad"],
+        ],
+    });
+    color = fields.Selection({
+        selection: [
+            ["white", "White"],
+            ["grey", "Grey"],
+            ["black", "Black"],
+        ],
+    });
+    _records = [
+        {
+            id: 1,
+            display_name: "first record",
+            is_raining_outside: false,
+            mood: "happy",
+            color: "white",
+        },
+    ];
+}
+
+defineModels([Partner]);
+
+test("badge selection field with filter, empty list", async () => {
+    await mountView({
+        type: "form",
+        resModel: "partner",
+        resId: 1,
+        arch: /* xml */ `
+            <form>
+                <field name="color" widget="selection_badge_with_filter"
+                       context="{'allowed_selection': []}" />
+            </form>
+        `,
+    });
+
+    expect(".o_selection_badge").not.toBeVisible();
```

Comment:
General minor remark: we rather do
```suggestion
    expect(".o_selection_badge").toHaveCount(0);
```
because with .not.toBeVisible(), we think that the element is in the dom but invisible, which isn't the case (and which we don't want actually)